transport/src/main/java/io/camunda/zeebe/transport/stream/impl/messages/ErrorResponse.java

```diff
@@ -100,4 +101,20 @@ public boolean equals(final Object o) {
   public String toString() {
     return "ErrorResponse{" + "message=" + message() + ", code=" + code + '}';
   }
+
+  public StreamResponseException asException() {
+    return new StacklessException(this);
+  }
+
+  private static final class StacklessException extends StreamResponseException {
```

Comment:
@Zelldon - was curious if you'd say something here :smile: My idea was that with exceptions, the "expensive" part is building the stack trace (see https://www.baeldung.com/java-exceptions-performance with reproducible benchmarks)

Reply:
Interesting. I somehow have over scrolled this 🤔 but yeah makes sense for me in this usecase.

Reply:
It's for compatibility with the `CompletableFuture` API to differentiate with `complete` and `completeExceptionally`. If we used `Either` or something instead then we could use different types.